packages/block-library/src/paragraph/edit.native.js

```diff
@@ -71,6 +72,20 @@ class ParagraphEdit extends Component {
 		}
 	}
 
+	onReplace( blocks ) {
```

Comment:
This seems like fully copy pasted from the web side module so, let's extract it into its own module and resuse it both on the web and native. What do you think?

Reply:
I had a similar thought, my initial hesitation was that the method calls a reference to the parent component method of the same name, and the shadowed (inner scope) reference is to the parent version, so I wasn't sure if that would be easy to follow in a separate file. The more I think on it, your suggestion makes sense. I'll try it and push a commit.

Reply:
I have created [this commit](https://github.com/mkevins/gutenberg/commit/1fcd88c134f9094bdb227b5b273dd833cc5ac011) on a new branch which refactors this. After creating it, I'm conflicted, as I wonder about the pattern introduced. Originally, the function is defined as a prototype method, but is bound to `this` in the constructor (a common pattern, but effectively making it a new function, occupying its own place in memory). With this commit, it will also need be bound to `this` in the constructor, so I don't think there will be any impact on memory. However, it differs from the pattern used elsewhere, where we refactor stand-alone functions that were already being exported from an index file (whereas here we are refactoring an internal prototype method).

Having `onReplace` as a copy-pasted prototype method in both the vanilla and native versions (i.e. leaving it as it is) is not DRY, so modifications / maintenance requires more work. On the other hand, I'm uncertain whether changes to the vanilla implementation would always be safe for the native side, so maybe there is some benefit to the manual effort required to mirror such modifications / changes. I don't have a strong opinion one way or the other, but I just wanted to document my thoughts as we may encounter similar situations in other areas of the codebases. WDYT?

A few notes: 
- I had to use an alias for parent onReplace function in the standalone file to satisfy linting rules, but I think this actually clarifies the code.
- I had to remove onReplace from some of the destructured assignments, and use the more verbose `this.props.onReplace` for the same linting reasons.

Reply:
I see, thanks for drilling down on this @mkevins! It seems that extracting this function ends up complicating the code, perhaps because the function is rather trivial in the first place.

I had a look at the reason it was introduced ([see this commit](https://github.com/WordPress/gutenberg/commit/a9142441669ab422338190311ee6fdc000ff5c2d)) and I think the function has little chances to get revised.

So, I think it should be OK to just have it copy-pasted from the web-side module, like you currently have it in the PR. Is that OK with you too?

Reply:
Yes, I'll leave it as it is.